src/db.c

```diff
@@ -1944,6 +1944,72 @@ static struct db_sys_list *_db_rule_gen_64(const struct arch_def *arch,
 	return NULL;
 }
 
+static int _db_rule_gen_arg_32(struct db_sys_list *s_new,
+			       const struct arch_def *arch,
+			       const struct db_api_arg *api_arg,
+			       struct db_arg_chain_tree *prev_nodes[4],
+			       bool *tf_flag)
+{
+	struct db_arg_chain_tree *c_iter = NULL;
+	struct db_arg_chain_tree **node;
+
+	if (api_arg->valid == 0)
+		return 0;
+
+	/* skip generating instructions which are no-ops */
+	if (!_db_arg_cmp_need_lo(api_arg))
+		return 0;
+
+	c_iter = zmalloc(sizeof(*c_iter));
+	if (c_iter == NULL)
+		return -1;
+
+	c_iter->arg = api_arg->arg;
+	c_iter->arg_h_flg = false;
+	c_iter->arg_offset = arch_arg_offset(arch, c_iter->arg);
+	c_iter->op = api_arg->op;
+	c_iter->op_orig = api_arg->op;
+	/* implicitly strips off the upper 32 bit */
+	c_iter->mask = api_arg->mask;
+	c_iter->datum = api_arg->datum;
+	c_iter->datum_full = api_arg->datum;
+
+	/* link in the new node and update the chain */
+	for (node = prev_nodes; *node != NULL; node++)
+		if (*tf_flag)
+			(*node)->nxt_t = _db_node_get(c_iter);
+		else
+			(*node)->nxt_f = _db_node_get(c_iter);
```

Comment:
I know braces ("{ ... }") aren't strictly required for this for-loop body, but please add them to help readability and prevent silly mistakes in the future.